main.go

```diff
@@ -150,43 +155,38 @@ func uploadFile(api *CacophonyAPI, filename string, info *cptvInfo) error {
 }
 
 type cptvInfo struct {
-	timestamp time.Time
-	duration  int
+	timestamp  time.Time
+	duration   int
+	devicename string
 }
 
 func extractCPTVInfo(filename string) (*cptvInfo, error) {
-	file, err := os.Open(filename)
+	f, err := os.Open(filename)
 	if err != nil {
 		return nil, err
 	}
-	defer file.Close()
+	defer f.Close()
 
-	// TODO: use the higher level cptv.Reader type (when it exists!)
-	p, err := cptv.NewParser(bufio.NewReader(file))
-	if err != nil {
-		return nil, err
-	}
-	fields, err := p.Header()
-	if err != nil {
-		return nil, err
-	}
-	timestamp, err := fields.Timestamp(cptv.Timestamp)
+	r, err := cptv.NewReader(f)
 	if err != nil {
 		return nil, err
 	}
 
 	frames := 0
 	for {
-		_, _, err := p.Frame()
-		if err == io.EOF {
-			break
-		} else if err != nil {
+		frame := new(lepton3.Frame)
+		err := r.ReadFrame(frame)
```

Comment:
The problem here is that `cptv.Reader` is decompressing the fields which is a lot of extra work. 

Using the lower level approach for using `cptv.Parser` avoids decompression.

We should add a `FrameCount()` method to `cptv.Reader` to support this use case. For now, perhaps revert this part of the change to avoid the extra computational work.